WordPress/src/main/java/org/wordpress/android/viewmodel/history/HistoryViewModel.kt

```diff
@@ -81,30 +81,33 @@ class HistoryViewModel @Inject constructor(
     init {
         lifecycleRegistry.markState(Lifecycle.State.CREATED)
         dispatcher.register(this)
+    }
+
+    fun create(localPostId: Int, site: SiteModel) {
+        if (isStarted) {
+            return
+        }
+        isStarted = true
+        this.site = site
         connectionStatus.observe(this, Observer {
             if (it == AVAILABLE) {
                 fetchRevisions()
             }
         })
-    }
-
-    fun create(localPostId: Int, site: SiteModel) = launch {
-        if (isStarted) {
-            return@launch
-        }
-
-        val post: PostModel? = withContext(bgDispatcher) { postStore.getPostByLocalPostId(localPostId) }
+        launch {
```

Comment:
It seems the default dispatche is `Dispatchers.Default` - do you really want to perform all these operations in a background thread?

Reply:
Ohh I noticed that it was an existing behavior before these changes. I'm wondering how come it doesn't crash on `[liveadata].value = `. I thought it crashes when performed on a background thread 😕 .

Reply:
Ok ignore me... `ScopedViewModel` contains a variable called `defaultDispatcher` but we actually pass main dispatcher there... We might want to rename that variable as it's quite confusing:).